code/modules/ghostroles/spawner/base.dm

```diff
@@ -116,10 +126,33 @@
 
 	return null //If we dont have anything return null
 
+//Selects a spawnatom from the list of available atoms and removes it if use is set to true (default)
+/datum/ghostspawner/proc/select_spawnatom(var/use=TRUE)
+	if(loc_type != GS_LOC_ATOM)
+		log_debug("Ghostspawner: select_spawnatom is not valid for spawner [short_name] as it is not atom pased")
+		return null
+	var/atom/A = pick(spawn_atoms)
+	if(use)
+		spawn_atoms -= A
+	return A
+
 //The proc to actually spawn in the user
 /datum/ghostspawner/proc/spawn_mob(mob/user)
 	//OVERWRITE THIS IN THE CHILD IMPLEMENTATIONS to return the spawned in mob !!!
-	return null
+
+	//This is a basic proc for atom based spawners.
+	//  Location based spawners usually need a bit more logic
+	//  Atom based spanwers select a atom using the select_spawnatom() proc and then assigned a player to it
+	//  by calling assign_player() with the current mob of the player that should be assigned as a argument.
+	if(loc_type != GS_LOC_ATOM)
+		return null
```

Comment:
I feel like this should have a log

Reply:
This is the cant_see proc.
That is called whenever something needs to determine if a user can see a specific spawn atom.
Logging this would cause a flood of logs.